### PR TITLE
Added the search bar and sorted on the basis maximum project first

### PR DESCRIPTION
**Pull request by ditya0504**
Targets #2627 

### Description
I'm using the searching function for employee name and sorted on the basis of maximum projects first.
### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

File: Modules/HR/Http/Controllers/EmployeeController.php [hunk 1]
```diff
@@ -28,10 +28,24 @@ public function __construct(EmployeeService $service)
...
     {
+        $this->authorize('list', Employee::class);
+        $search = request()->query('employeename') ?: '';
+        $employeeData = Employee::with('employees');
         $filters = $request->all();
         $filters = $filters ?: $this->service->defaultFilters();
         $name = request('name');
-        $employeeData = Employee::where('staff_type', $name)->get();
+        $employeeData = Employee::where('staff_type', $name)
+            ->leftJoin('project_team_members', 'employees.user_id', '=', 'project_team_members.team_member_id')
```
Comment (GauravGusain98) [inline Modules/HR/Http/Controllers/EmployeeController.php:38]: Here, I think we have to exclude the projects in which the user is not a part now but was earlier. For this you have to add a check where ended_on is NOT NULL. Then count the number of projects.

File: resources/views/hr/employees/index.blade.php [hunk 2]
```diff
@@ -53,7 +57,7 @@
                     @if($employee->user == null)
                         0
                     @else
-                        {{count($employee->user->activeProjectTeamMembers)}}
+                        {{$employee-> Count}}
```
Comment (GauravGusain98) [inline resources/views/hr/employees/index.blade.php:60]:
```suggestion
                        {{$employee->project_count}}
```

File: Modules/HR/Http/Controllers/EmployeeController.php [hunk 3]
```diff
@@ -28,10 +28,24 @@ public function __construct(EmployeeService $service)
...
+            ->selectRaw('employees.*, team_member_id, count(team_member_id) as Count')
```
Comment (GauravGusain98) [inline Modules/HR/Http/Controllers/EmployeeController.php:39]: Lets use more meaningful name
```suggestion
            ->selectRaw('employees.*, team_member_id, count(team_member_id) as project_count')
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
Please check the comments

File: Modules/HR/Http/Controllers/EmployeeController.php (hunk 1 above)
Comment (ditya0504) [reply Modules/HR/Http/Controllers/EmployeeController.php:38]: @GauravGusain98 Sorry but this was not mentioned in the issue, so i was not sure about this.

File: resources/views/hr/employees/index.blade.php (hunk 2 above)
Comment (ditya0504) [reply resources/views/hr/employees/index.blade.php:60]: ok will change this

File: Modules/HR/Http/Controllers/EmployeeController.php (hunk 1 above)
Comment (GauravGusain98) [reply Modules/HR/Http/Controllers/EmployeeController.php:38]: Actually it was in the code here
image

File: resources/views/hr/employees/index.blade.php [hunk 4]
```diff
@@ -53,7 +57,7 @@
                     @if($employee->user == null)
                         0
                     @else
-                        {{count($employee->user->activeProjectTeamMembers)}}
+                        {{$employee-> project_count}}
```
Comment (GauravGusain98) [inline resources/views/hr/employees/index.blade.php:60]:
```suggestion
                        {{$employee->project_count}}
```

File: resources/views/hr/employees/index.blade.php [hunk 5]
```diff
@@ -7,14 +7,18 @@
     <br><br>
     <div class="d-flex">
         <h1>{{request()->get('name')}} ({{count($employees)}})</h1>
-        <form id="employeeFilterForm">
+        <form id="employeeFilterForm" class ="d-md-flex justify-content-between ml-md-3">
```
Comment (GauravGusain98) [inline resources/views/hr/employees/index.blade.php:10]:
```suggestion
        <form id="employeeFilterForm" class="d-md-flex justify-content-between ml-md-3">
```

File: Modules/HR/Http/Controllers/EmployeeController.php (hunk 1 above)
Comment (GauravGusain98) [reply Modules/HR/Http/Controllers/EmployeeController.php:38]: please implement the changes

File: Modules/HR/Http/Controllers/EmployeeController.php [hunk 6]
```diff
@@ -28,10 +28,25 @@ public function __construct(EmployeeService $service)
...
+        $search = request()->query('employeename') ?: '';
+        $employeeData = Employee::with('employees');
         $filters = $request->all();
         $filters = $filters ?: $this->service->defaultFilters();
         $name = request('name');
-        $employeeData = Employee::where('staff_type', $name)->get();
+        $employeeData = Employee::where('staff_type', $name)
+            ->leftJoin('project_team_members', 'employees.user_id', '=', 'project_team_members.team_member_id')
+            ->selectRaw('employees.*, team_member_id, count(team_member_id) as project_count')
+            ->where('project_team_members.ended_on', '!=', 'NULL')
```
Comment (GauravGusain98) [inline Modules/HR/Http/Controllers/EmployeeController.php:40]: If the ended_on column is filled then it means that the team member is not a part of project
```suggestion
            ->where('project_team_members.ended_on', '=', 'NULL')
```

**Review by GauravGusain98 — CHANGES_REQUESTED** [review]
minor change requested, else looks good

File: Modules/HR/Http/Controllers/EmployeeController.php [hunk 7]
```diff
@@ -28,10 +28,25 @@ public function __construct(EmployeeService $service)
...
+        $search = request()->query('employeename') ?: '';
+        $employeeData = Employee::with('employees');
         $filters = $request->all();
         $filters = $filters ?: $this->service->defaultFilters();
         $name = request('name');
-        $employeeData = Employee::where('staff_type', $name)->get();
+        $employeeData = Employee::where('staff_type', $name)
+            ->leftJoin('project_team_members', 'employees.user_id', '=', 'project_team_members.team_member_id')
+            ->selectRaw('employees.*, team_member_id, count(team_member_id) as project_count')
+            ->WhereNull('project_team_members.ended_on')
```
Comment (GauravGusain98) [inline Modules/HR/Http/Controllers/EmployeeController.php:40]:
```suggestion
            ->whereNull('project_team_members.ended_on')
```

**Review by GauravGusain98 — APPROVED** [review]
minor change. Fixing it. Else looks good

**Pull request merged**